tests/queries/0_stateless/02885_create_distributed_table_without_as.sql

```diff
@@ -0,0 +1,16 @@
+-- Here a Distributed table without AS must detect its structure.
+
+DROP TABLE IF EXISTS dist_tbl;
+DROP TABLE IF EXISTS local_tbl;
+
+CREATE TABLE local_tbl (`key` UInt32, `value` UInt32 DEFAULT 42) ENGINE = MergeTree ORDER BY key;
+CREATE TABLE dist_tbl ENGINE = Distributed('test_shard_localhost', currentDatabase(), 'local_tbl', rand());
+SHOW CREATE TABLE dist_tbl;
```

Comment:
Before this fix this line output

```
CREATE TABLE default.dist_tbl (key UInt32, value UInt32) ENGINE = Distributed('test_shard_localhost', 'default', 'local_tbl', rand())
```

and then `INSERT` didn't apply defaults and `SELECT * FROM local_tbl` showed
```
99      0
```